src/WaCore.Web/Middleware/WebApiExceptionHandler/WacWebApiExceptionHandlerMiddleware.cs

```diff
@@ -0,0 +1,95 @@
+﻿using System;
+using System.IO;
+using System.Net;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Serialization;
+using WaCore.Common.ExceptionHandling;
+using WaCore.Web.Exceptions;
+
+namespace WaCore.Web.Middleware.WebApiExceptionHandler
+{
+    /// <summary>
+    /// A middleware to handle exceptions in Web Api project and generate HTTP response based on exception type.
+    /// By default this middleware will handle all exceptions inherited from <see cref="WebApiException"/>.
+    /// </summary>
+    public class WacWebApiExceptionHandlerMiddleware
+    {
+        private const string UnhandledExceptionMessage = "Internal server error";
```

Comment:
I think it would be nice to put this property in the `WacWebApiExceptionHandlerOptions` class, so the user can change the text, if he wants to. (`public string UnhandledExceptionMessage = "Internal server error";`)